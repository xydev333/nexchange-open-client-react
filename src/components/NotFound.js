import React from 'react';
import { I18n } from 'react-i18next';
import '../css/not-found.scss';

const NotFound = () => (
  <div id="not-found" className="text-center">
	<I18n ns="translations">
	{(t) => (
		<h1>404</h1>
		<h2>{t('error.notfound')}</h2>
     )}
    </I18n>
  </div>
);

export default NotFound;
