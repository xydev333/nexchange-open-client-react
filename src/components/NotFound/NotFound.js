import React, {Component} from 'react';
import { I18n } from 'react-i18next';
import { withRouter } from 'react-router-dom';
import styles from  './NotFound.scss';

export class NotFound extends Component {

  componentDidMount() {
    $("#root").css({'padding-bottom': "0px"});
  }

  componentWillUnmount() {
    $("#root").css({'padding-bottom': "114px"});
  }

  render() {
   return (<I18n ns="translations">
    {t => (
      <div id="not-found" className={styles.container}>
        <div className="col-xs-12 col-sm-12 col-md-5 col-md-offset-1 col-lg-5 col-lg-offset-1">
          <h2>{t('error.notfound1')}</h2>
          <h5>{t('error.notfound21')} <strong>{t('error.notfound22')}</strong> {t('error.notfound23')}</h5>
          <a onClick={this.props.history.goBack}><i className="fas fa-long-arrow-alt-left"></i>{t('error.notfound22')}</a>
        </div>
      </div>
    )}
    </I18n>);
  }  
}

export default withRouter(NotFound);
