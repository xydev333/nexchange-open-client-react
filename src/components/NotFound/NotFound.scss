.container {
    position: relative;
    height: 100vh;
    display: flex;
    align-items: center;
    color: #fff;
    background: -moz-linear-gradient(45deg, rgba(2, 3, 31, 1) 0%, rgba(2, 3, 31, 1) 44%, rgba(26, 157, 139, 1) 100%); /* ff3.6+ */
    background: -webkit-gradient(
      linear,
      left bottom,
      right top,
      color-stop(0%, rgba(2, 3, 31, 1)),
      color-stop(44%, rgba(2, 3, 31, 1)),
      color-stop(100%, rgba(26, 157, 139, 1))
    ); /* safari4+,chrome */
    background: -webkit-linear-gradient(
      45deg,
      rgba(2, 3, 31, 1) 0%,
      rgba(2, 3, 31, 1) 44%,
      rgba(26, 157, 139, 1) 100%
    ); /* safari5.1+,chrome10+ */
    background: -o-linear-gradient(45deg, rgba(2, 3, 31, 1) 0%, rgba(2, 3, 31, 1) 44%, rgba(26, 157, 139, 1) 100%); /* opera 11.10+ */
    background: -ms-linear-gradient(45deg, rgba(2, 3, 31, 1) 0%, rgba(2, 3, 31, 1) 44%, rgba(26, 157, 139, 1) 100%); /* ie10+ */
    background: linear-gradient(45deg, rgba(2, 3, 31, 1) 0%, rgba(2, 3, 31, 1) 44%, rgba(26, 157, 139, 1) 100%); /* w3c */
    filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#1A9D8B', endColorstr='#02031f',GradientType=1 ); /* ie6-9 */
  
    @media (min-height: $screen-lg-height) and (min-width: $screen-md-min) {
      min-height: 100vh;
    }
  
    @media (min-width: $screen-md-min) {
      background-color: #01031e;
      background-image: url('../../img/ellipse.svg');
      background-position: 130% 50%;
      background-size: 150%;
      background-repeat: no-repeat;
    }

    h5 {
        font-size: 18px;
    }

    a {
        font-size:  18px;
        color: #fff;
        cursor: pointer;
        i {
            margin-right: 10px;
        }
        &:hover {
            color: #fff;
            text-decoration: none;
        }
    }
}