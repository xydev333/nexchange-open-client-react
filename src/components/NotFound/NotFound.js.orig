import React from 'react';
<<<<<<< HEAD:src/components/NotFound/NotFound.js
import './NotFound.css';

const NotFound = () => (
  <div className="text-center">
    <h1>404</h1>
    <h2>Page not found :(</h2>
=======
import { I18n } from 'react-i18next';
import '../css/not-found.scss';

const NotFound = () => (
  <div id="not-found" className="text-center">
	<I18n ns="translations">
	{(t) => (
		<h1>404</h1>
		<h2>{t('error.notfound')}</h2>
     )}
    </I18n>
>>>>>>> Start merging i18n integration:src/components/NotFound.js
  </div>
);

export default NotFound;
