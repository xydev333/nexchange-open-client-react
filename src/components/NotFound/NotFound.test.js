import React from 'react';
import { shallow } from 'enzyme';
import { NotFound } from './NotFound';

describe('NotFound', () => {
  it('renders correctly', () => {
    const goBack = jest.fn();
    const wrapper = shallow(<NotFound history={goBack} />);
    expect(wrapper).toMatchSnapshot();
  });
  // it('should call history.push', () => {
  //     const wrapper = mount(<MyComponent />);
  //     // perform your test
  // });
});
