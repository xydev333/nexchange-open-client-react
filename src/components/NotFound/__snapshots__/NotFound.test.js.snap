// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
  >
    [Function]
  </I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          404
        </h1>,
        <h2>
          error.notfound
        </h2>,
      ],
      "className": "text-center",
      "id": "not-found",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "404",
        },
        "ref": null,
        "rendered": "404",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "error.notfound",
        },
        "ref": null,
        "rendered": "error.notfound",
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            404
          </h1>,
          <h2>
            error.notfound
          </h2>,
        ],
        "className": "text-center",
        "id": "not-found",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "404",
          },
          "ref": null,
          "rendered": "404",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "error.notfound",
          },
          "ref": null,
          "rendered": "error.notfound",
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {},
  },
}
`;
